piplicenses/output/CSVPrettyTable.py

```diff
@@ -0,0 +1,141 @@
+
+        All inputs are treated as strings or byte sequences. Byte inputs are
+        decoded as UTF-8, with fallbacks for characters unrepresentable in UTF-8.
+
+        Args:
+            **kwargs: Keyword arguments to customize row selection and formatting.
+                These are passed through to PrettyTable's internal `_get_options`
+                and `_get_rows` methods. Typical options include:
+                - `fields`: A list of field names to include.
+                - `start`: The starting row index.
+                - `end`: The ending row index.
+                - `header`: Whether to include the header row.
+                Refer to PrettyTable documentation for full options.
+
+        Returns:
+            A string representing the entire table in CSV format. Each value is
+            enclosed in double quotes, with internal double quotes escaped by
+            doubling ("" -> ""\""). Rows are separated by newline characters.
+
+        Example:
+            >>> table = CSVPrettyTable()
+            >>> table.field_names = ["Name", "Age"]
+            >>> table.add_row(["Alice", 30])
+            >>> table.add_row(['Bob "The Builder"', 25])
+            >>> print(table.get_string())
+            "Name","Age"
+            "Alice","30"
+            "Bob ""The Builder"\"","25"
+        """
+
+        def esc_quotes(val: bytes | str) -> str:
+            """
+            Escape double quotes in a value according to RFC 4180.
+
+            This helper function ensures that any double-quote characters in the
+            cell value are represented as two consecutive double quotes.
+            It accepts either bytes or string inputs.
+
+            Args:
+                val: The cell value, either as a str or bytes.
+
+            Returns:
+                A string with all double quotes escaped (each " becomes "").
+
+            Raises:
+                UnicodeDecodeError: If a bytes input cannot be decoded as UTF-8
+                                   (covered by fallback logic).
+                UnicodeEncodeError: If a str cannot be encoded in the fallback
+                                    path (covered by fallback logic).
+
+            References:
+                https://tools.ietf.org/html/rfc4180
+            """
+            try:
+                # If val is str, replace directly.
+                return cast(str, val).replace('"', '""')
+            except UnicodeDecodeError:  # pragma: no cover
+                # If val is bytes but contained invalid UTF-8, decode and then replace.
+                return cast(bytes, val).decode("utf-8").replace('"', '""')
+            except UnicodeEncodeError:  # pragma: no cover
+                # Fallback: escape to unicode_escape and then replace quotes.
+                # Why is this still attempting strings? This should probably replace(b'"', b'""')
+                return str(cast(str, val).encode("unicode_escape").replace('"', '""'))  # type: ignore[arg-type]
```

Comment:
Consider after encoding this is `bytes.replace()` and should probably use byte strings (albeit it is then cast back to a string after the replace):
```suggestion
                return str(cast(str, val).encode("unicode_escape").replace(b'"', b'""'))  # type: ignore[arg-type]
```